aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -441,6 +444,150 @@ std::tuple<Tensor&,Tensor&> min_out(Tensor& min, Tensor& min_indices,
   return result;
 }
 
+static Tensor scalar_tensor_like(const Tensor& like, const Scalar& value) {
+  // FIXME: This should do type promotion with value, but that would be a BC-break
+  return at::scalar_tensor(value, at::device(like.device()).dtype(like.scalar_type()));
+}
+
+static c10::optional<Tensor> scalar_tensor_like(const Tensor& like, const c10::optional<Scalar>& value) {
+  if (value) {
+    return scalar_tensor_like(like, *value);
+  }
+  return c10::nullopt;
+}
+
+Tensor& clamp_out(const Tensor& self, c10::optional<Scalar> min, c10::optional<Scalar> max, Tensor& result) {
+  return at::clamp_outf(self, scalar_tensor_like(self, min), scalar_tensor_like(self, max), result);
+}
+
+Tensor& clamp_out(const Tensor& self, const c10::optional<Tensor>& min,
+                  const c10::optional<Tensor>& max, Tensor& result) {
+  if (min && max) {
+    TORCH_CHECK(self.layout() == Layout::Strided,
+                "clamp only supports strided layout, got: ", self.layout());
+    auto iter = TensorIteratorConfig()
+                .set_check_mem_overlap(true)
+                .add_output(result)
+                .add_input(self)
+                .add_input(*min)
+                .add_input(*max)
+                .promote_inputs_to_common_dtype(true)
+                .cast_common_dtype_to_outputs(true)
+                .enforce_safe_casting_to_output(true)
+                .build();
+    clamp_stub(iter.device_type(), iter);
+  } else if (max) {
+    at::clamp_max_outf(self, *max, result);
+  } else if (min) {
+    at::clamp_min_outf(self, *min, result);
+  } else {
+    TORCH_CHECK(false, "At least one of 'min' or 'max' must not be None");
```

Comment:
Start error messages with the name of the function they originate from, "torch.clamp(): at least..."